apps/server/src/modules/school/uc/school.uc.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@nestjs/common';
+import { SchoolDto } from '@src/modules/school/uc/dto/school.dto';
+import { SchoolService } from '@src/modules/school/service/school.service';
+import { ProvisioningSchoolOutputDto } from '@src/modules/provisioning/dto/provisioning-school-output.dto';
+import { SchoolUcMapper } from '@src/modules/school/mapper/school.uc.mapper';
+
+@Injectable()
+export class SchoolUc {
+	constructor(readonly schoolService: SchoolService) {}
+
+	async saveProvisioningSchoolOutputDto(schoolDto: ProvisioningSchoolOutputDto): Promise<SchoolDto> {
```

Comment:
I really like having a seperate uc for the case of creating a school from the provisioning, that translates between the "language" of the provisioning service and the school uc.
However, I think this should actually be part of the provisioning module, so that the school module does not need to know that the provisioning exists. In the future, I would expect the same for other objects that are created as well.

Reply:
I argue that it is better to have the Domains translate provisioning DTOs into domain specific DTOs. Having this logic in the provisioning module could cause any change in the receiving Uc have changes in the provisioning module and cause side effects. 

Another Option we could discuss is having shared transactional DTOs that are shared by all modules and have to be used soley for calls between Ucs. We can discuss this topic further in the Arch Round if desired

Reply:
So we can resolve this?

Reply:
I second @Metauriel
The school module offers an API to create and update schools and the provisioning module is using that API. So the provisioning module needs to handle the mapping. The schools module should not even know provisioning exists.